src/engine.rs

```diff
@@ -2230,4 +2285,128 @@ mod tests {
             assert!(engine.file_span(LogQueue::Append).0 > start);
         }
     }
+
+    #[test]
+    fn test_start_engine_without_recycle() {
+        let dir = tempfile::Builder::new()
+            .prefix("test_start_without_recycle")
+            .tempdir()
+            .unwrap();
+        let path = dir.path().to_str().unwrap();
+        let cfg = Config {
+            dir: path.to_owned(),
+            enable_log_recycle: false,
+            ..Default::default()
+        };
+        let file_system = Arc::new(DeleteMonitoredFileSystem::new());
+        let engine =
+            RaftLogEngine::open_with_file_system(cfg.clone(), file_system.clone()).unwrap();
+        let (start, _) = engine.file_span(LogQueue::Append);
+        // Only one valid file left, the last one => active_file.
+        assert_eq!(engine.file_count(Some(LogQueue::Append)), 1);
+        assert_eq!(file_system.inner.file_count(), engine.file_count(None));
+        // Append data.
+        let entry_data = vec![b'x'; 512];
+        for rid in 1..=5 {
+            engine.append(rid, 11, 20, Some(&entry_data));
+        }
+        assert_eq!(engine.file_span(LogQueue::Append).0, start);
+        let (start, end) = engine.file_span(LogQueue::Append);
+        assert_eq!(file_system.inner.file_count(), engine.file_count(None));
+        drop(engine);
+        // Reopen the engine with a smaller capacity.
```

Comment:
Try to use newlines more often to make the code clearer.
```suggestion
        drop(engine);

        // Reopen the engine with a smaller capacity.
```